ironic-common.sh

```diff
@@ -0,0 +1,29 @@
+PROVISIONING_INTERFACE=${PROVISIONING_INTERFACE:-"provisioning"}
+
+# Wait for the interface or IP to be up, sets $IRONIC_IP
+function wait_for_interface_or_ip() {
+  # If $PROVISIONING_IP is specified, then we wait for that to become available on an interface, otherwise we look at $PROVISIONING_INTERFACE for an IP
+  if [ ! -z "${PROVISIONING_IP}" ];
+  then
+    until [ ! -z "${IRONIC_IP}" ]; do
+      echo "Waiting for ${PROVISIONING_IP} to be configured on an interface"
+      IRONIC_IP=$(ip -br addr show | grep "${PROVISIONING_IP}" | grep -Po "[^\s]+/[0-9]+" | sed -e 's%/.*%%' | head -n 1)
+      sleep 1
+    done
+  else
+    until [ ! -z "${IRONIC_IP}" ]; do
+      echo "Waiting for ${PROVISIONING_INTERFACE} interface to be configured"
+      IRONIC_IP=$(ip -br addr show dev $PROVISIONING_INTERFACE | grep -Po "[^\s]+/[0-9]+" | grep -e "^fd" -e "\." | sed -e 's%/.*%%' | head -n 1)
+      sleep 1
+    done
+
+    # If the IP contains a colon, then it's an IPv6 address, and the HTTP
+    # host needs surrounding with brackets
+    if [[ "$IRONIC_IP" =~ .*:.* ]]
+    then
+      IRONIC_URL_HOST="[$IRONIC_IP]"
+    else
+      IRONIC_URL_HOST=$IRONIC_IP
+    fi
+  fi
```

Comment:
Travelling at the moment so didn't have a change to test but looks like this if/else should end before we check $IRONIC_IP for IP version?

Reply:
You are right! There was already a fix for this IIRC. Will cherry-pick that here.

Reply:
Fixed

Reply:
Sounds like this ended up more than one cherry-pick, should we reference e.g https://github.com/metal3-io/ironic-inspector-image/pull/42 and any other PRs in the commit message?

Reply:
Sure, fixed.